app/pages/admin/featured-project-toggle.jsx

```diff
@@ -0,0 +1,46 @@
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import AutoSave from '../../components/auto-save';
+
+class FeaturedProjectToggle extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      project: props.project
```

Comment:
`Object.assign({}, props.project)` here, otherwise changes to `state.project` which actually change `props.project`.

This also means that `state.project` in this component will now be out of sync with `state.project` in the parent, so you'll need to pass the updated project back to the parent.

Updating something in state in two places might not be a great idea, but the current implementation might be limited by the `AutoSave` component, so this isn't a blocking issue.

Reply:
I feel like we went full circle. Which one do you think would be the best option in this case?

Reply:
Updating state in the parent seems like the ideal here, no, unless I'm missing something? Would that involve removing the `AutoSave`? If so, don't worry about it.

Reply:
```jsx
<FeaturedProjectToggle project={this.state.project} onChange={onProjectChange} />
```

Reply:
OK, I've got a proper working controlled component of that form. However the test now is useless. I've also tried to test the handler is called in the parent component, but that unsurprising fails. Should I test just the change of props in the controlled? I'm thinking of deleting its spec file and just leave the rendering test in the parent.

Reply:
Isn't the test that `props.onChange` is called when the box is toggled? I'm not sure I understand.